File: api-certvalidation/csharp/src/FinderIteration.cs
using tr.gov.tubitak.uekae.esya.api.asn.x509;

namespace tr.gov.tubitak.uekae.esya.api.certificate.validation
{
    /**
     * Iteration for certificate finding 
     */
    public abstract class FinderIteration
    {
        public abstract bool nextIteration(ValidationSystem aValidationSystem, ECertificate iCertificate);

        protected abstract bool _nextSource(ValidationSystem aValidationSystem, ECertificate iCertificate);

    }
}

File: api-certvalidation/src/tr/gov/tubitak/uekae/esya/api/certificate/validation/FinderIteration.java
package tr.gov.tubitak.uekae.esya.api.certificate.validation;

import tr.gov.tubitak.uekae.esya.api.asn.x509.ECertificate;
import tr.gov.tubitak.uekae.esya.api.common.ESYAException;

/**
 * Iteration for certificate finding 
 */
public abstract class FinderIteration {

	public abstract boolean nextIteration(ValidationSystem aValidationSystem, ECertificate iCertificate) throws ESYAException;

	protected abstract boolean _nextSource(ValidationSystem aValidationSystem, ECertificate iCertificate) throws ESYAException;

}

